### PR TITLE
BmapHelpers.py: fix [Errno 25] Inappropriate ioctl for device

### PR DESCRIPTION
**Pull request by tcler**
see detail:
 https://github.com/01org/bmap-tools/issues/29

Signed-off-by: Jianhong Yin <yin-jianhong@163.com>

### CODE REVIEW

File: bmaptools/BmapHelpers.py [hunk 1]
```diff
@@ -62,9 +62,10 @@ def get_block_size(file_obj):
 
     # Get the block size of the host file-system for the image file by calling
     # the FIGETBSZ ioctl (number 2).
-    binary_data = ioctl(file_obj, 2, struct.pack('I', 0))
-    bsize = struct.unpack('I', binary_data)[0]
-    if not bsize:
+    try:
+        binary_data = ioctl(file_obj, 2, struct.pack('I', 0))
+        bsize = struct.unpack('I', binary_data)[0]
+    except IOError as err:
```
Comment (kad) [inline bmaptools/BmapHelpers.py]: This would not work correctly in case if ioctl() returns successfully, but with bsize=0.

Comment (tcler) [reply bmaptools/BmapHelpers.py]: good point.
is it possible bsize == 0 if ioctl returns successfully ? if yes we need another patch

Comment (kad) [reply bmaptools/BmapHelpers.py]: Yes, it is possible in some specific cases. e.g. AUFS in some docker setups.

**Comment by dedekind** [comment]
Alexander, would you merge this possibly with some amendments? Better than having it hanging here.

**Pull request merged**